test/source/mock/fes/shared-tenant-fes-endpoints.ts

```diff
@@ -5,109 +5,154 @@ import { IncomingMessage } from 'http';
+  | 'DISABLE_FES_ACCESS_TOKEN'
+  | 'SETUP_ENSURE_IMPORTED_PRV_MATCH_LDAP_PUB';
+
+export type FesClientConfiguration = {
+  flags?: FesClientConfigurationFlag[];
+  custom_keyserver_url?: string;
+  key_manager_url?: string;
+  in_memory_pass_phrase_session_length?: number;
+  allow_attester_search_only_for_domains?: string[];
+  disallow_attester_search_for_domains?: string[];
+  enforce_keygen_algo?: string;
+  enforce_keygen_expire_months?: number;
+  allow_keys_openpgp_org_search_only_for_domains?: string[];
+  disallow_keys_openpgp_org_search_for_domains?: string[];
+};
+
+export interface FesConfig {
+  returnError?: { code: number; message: string };
+  clientConfiguration?: FesClientConfiguration;
+}
 
 const issuedAccessTokens: string[] = [];
-export const mockSharedTenantFesEndpoints: HandlersDefinition = {
-  // shared tenant fes location at https://flowcrypt.com/shared-tenant-fes/
-  '/shared-tenant-fes/api/': async ({}, req) => {
-    if (req.method === 'GET') {
+export const getMockSharedTenantFesEndpoints = (config: FesConfig | undefined): HandlersDefinition => {
+  return {
+    // shared tenant fes location at https://flowcrypt.com/shared-tenant-fes/
+    '/shared-tenant-fes/api/': async ({}, req) => {
+      if (req.method === 'GET') {
+        return {
+          vendor: 'Mock',
+          service: 'external-service',
+          orgId: 'standardsubdomainfes.test',
+          version: 'MOCK',
+          apiVersion: 'v1',
+        };
+      }
+      const port = parsePort(req);
+      if (req.headers.host === `fes.localhost:${port}`) {
+        // test `status404 does not return any fesUrl` uses this
+        // this makes enterprise version tolerate missing FES - explicit 404
+        throw new HttpClientErr(`Not found`, 404);
+      }
+      if (req.headers.host === `fes.google.mock.localhost:${port}`) {
+        // test `compose - auto include pubkey is inactive when our key is available on Wkd` uses this
+        // this makes enterprise version tolerate missing FES - explicit 404
+        throw new HttpClientErr(`Not found`, 404);
+      }
+      throw new HttpClientErr(`Not running any FES here: ${req.headers.host}`);
+    },
+    '/shared-tenant-fes/api/v1/client-configuration': async ({}, req) => {
+      // individual ClientConfiguration is tested using FlowCrypt backend mock, see BackendData.getClientConfiguration
+      if (req.method !== 'GET') {
+        throw new HttpClientErr('Unsupported method');
+      }
+      if (config) {
+        if (config.returnError) {
+          throw new HttpClientErr(config.returnError.message, config.returnError.code);
+        }
+        return {
+          clientConfiguration: config.clientConfiguration,
+        };
+      }
       return {
-        vendor: 'Mock',
-        service: 'external-service',
-        orgId: 'standardsubdomainfes.test',
-        version: 'MOCK',
-        apiVersion: 'v1',
+        clientConfiguration: {
+          flags: [],
+        },
```

Comment:
@sosnovsky, In `Attester`, we returned an error when `config` was not defined, as shown below.
However, for `FES`, I think it would be better to return an empty client configuration instead of throwing an error. Otherwise, we would have to add `clientConfiguration: { flags: [] }` to every test, which would be duplicative and not provide much benefit.

What are your thoughts on this?

```ts
throw new HttpClientErr(`Not implemented: ${req.method}`);
```

Reply:
Sounds good, let's do it this way here

Reply:
OK